src/CoreWCF.AzureQueueStorage/src/CoreWCF/Configuration/ServiceCollectionExtension.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.Storage.Queues;
+using CoreWCF.Channels;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace CoreWCF.Configuration
+{
+    public static class ServiceCollectionExtension
+    {
+        public static IServiceCollection AddServiceModelAzureQueueStorageSupport(this IServiceCollection services)
+        {
+            services.AddSingleton<QueueClient, QueueClient>();
+            return services;
```

Comment:
When the DI requested service type and the implementation type are the same, you can just use `services.AddSingleton<TService>();` I'm also not seeing where you are requesting QueueClient from DI.

Reply:
Thank you for the tip. I have the updated the code now to request the client from DI.